pkg/monitortests/etcd/etcdloganalyzer/monitortest.go

```diff
@@ -141,8 +146,45 @@ func (*etcdLogAnalyzer) ConstructComputedIntervals(ctx context.Context, starting
 	return ret, nil
 }
 
-func (*etcdLogAnalyzer) EvaluateTestsFromConstructedIntervals(ctx context.Context, finalIntervals monitorapi.Intervals) ([]*junitapi.JUnitTestCase, error) {
-	return nil, nil
+func (w *etcdLogAnalyzer) EvaluateTestsFromConstructedIntervals(ctx context.Context, finalIntervals monitorapi.Intervals) ([]*junitapi.JUnitTestCase, error) {
+	etcdIntervals := monitorapi.Intervals{}
+	for _, interval := range finalIntervals {
+		value, ok := interval.Message.Annotations[monitorapi.AnnotationConstructed]
+		if !ok {
+			continue
+		}
+		if value != monitorapi.ConstructionOwnerEtcdLifecycle {
+			continue
+		}
+		if len(interval.Message.Reason) != 0 {
+			continue
+		}
+		if interval.Locator.HasKey("node") {
+			if len(interval.Locator.Keys["node"]) == 0 {
+				continue
+			}
+		}
+		if value, ok := interval.Message.Annotations[monitorapi.AnnotationEtcdLeader]; ok && len(value) == 0 {
+			continue
+		}
+		etcdIntervals = append(etcdIntervals, interval)
+	}
+	sort.Sort(etcdIntervals)
+
+	junitTest := &junitTest{
+		name:     fmt.Sprintf("[sig-etcd] cluster should not be without a leader for more than %s seconds", leaderlessTimeout),
```

Comment:
ITYM
```suggestion
		name:     fmt.Sprintf("[sig-etcd] cluster should not be without a leader for more than %s", leaderlessTimeout),
```

but actually in the event that you change the threshold I don't think you want it in the name at all (you would need to remap the test to be considered "the same" or mark the old name as deliberately removed).

```suggestion
		name:     "[sig-etcd] cluster should not be without a leader for too long",
```

Reply:
done 👍🏽